frontend/src/app/shared/regex.utils.ts

```diff
@@ -0,0 +1,224 @@
+// all base58 characters
+const BASE58_CHARS = `[a-km-zA-HJ-NP-Z1-9]`;
+
+// all bech32 characters (after the separator)
+const BECH32_CHARS_LW = `[ac-hj-np-z02-9]`;
+const BECH32_CHARS_UP = `[AC-HJ-NP-Z02-9]`;
+
+// Hex characters
+const HEX_CHARS = `[a-fA-F0-9]`;
+
+// A regex to say "A single 0 OR any number with no leading zeroes"
+// Capped at 13 digits so as to not be confused with lightning channel IDs (which are around 17 digits)
+// (?:             // Start a non-capturing group
+//   0             // A single 0
+//   |             // OR
+//   [1-9][0-9]{0,12} // Any succession of numbers up to 13 digits starting with 1-9
+// )               // End the non-capturing group.
+const ZERO_INDEX_NUMBER_CHARS = `(?:0|[1-9][0-9]{0,12})`;
```

Comment:
this could match a lightning channel id (in regtest). a channel id can be anywhere from 2^40 (no channels can be confirmed in the genesis block, so height > 0) to 2^64-1, with the lowest being 13 characters long, this regex is used for blockheight and transaction output index, which are both limited to 32 bit integers. the highest 32 bit int is only 10 characters long